server/cluster/topology_manager.go

```diff
@@ -243,6 +246,50 @@ func (m *TopologyManagerImpl) RemoveTable(ctx context.Context, tableID storage.T
 	}, nil
 }
 
+func (m *TopologyManagerImpl) UpdateShardTables(ctx context.Context, shardTables ShardTables) (ShardVersionUpdate, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	if _, ok := m.shardNodesMapping[shardTables.Shard.ID]; !ok {
+		return ShardVersionUpdate{}, ErrShardNotFound.WithCausef("shard id:%d", shardTables.Shard.ID)
+	}
+
+	prevVersion := uint64(0)
+	shardView, ok := m.shardTablesMapping[shardTables.Shard.ID]
+	if ok {
```

Comment:
If ok is false, the `prevVersion` is wrong?
Shall we return error here.

Reply:
`ShardView` is not exists when it has not been created, this is a normal condition, and error cannot be returned.